### PR TITLE
Update other pnp sample to use native APIs

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: sdk/samples/iot/paho_iot_pnp_sample.c [hunk 1]
```diff
@@ -796,36 +698,36 @@ static void handle_command_request(
...
     az_iot_status status,
     az_span response)
 {
   az_result rc;
 
-  // Get the Methods response topic to publish the command response.
-  char methods_response_topic_buffer[128];
-  rc = az_iot_hub_client_methods_response_get_publish_topic(
-      &hub_client,
+  // Get the command response topic to publish the command response.
```
Comment (momuno) [inline sdk/samples/iot/paho_iot_pnp_sample.c:713]: nit: commands plural
here and elsewhere for commands response topic (basing off function name az_iot_pnp_client_commands_response_get_publish_topic)

**Review by momuno — APPROVED** [review]

File: sdk/samples/iot/paho_iot_pnp_sample.c [hunk 2]
```diff
@@ -328,57 +316,58 @@ static void subscribe_mqtt_client_to_iot_hub_topics(void)
...
-      sizeof(twin_document_topic_buffer),
+      property_document_topic_buffer,
+      sizeof(property_document_topic_buffer),
       NULL);
 
   if (az_result_failed(rc))
   {
-    IOT_SAMPLE_LOG_ERROR("Failed to get the Twin Document topic: az_result return code %04x", rc);
+    IOT_SAMPLE_LOG_ERROR(
+        "Failed to get the property document topic: az_result return code %04x", rc);
```
[the remark this answers is not in the document]
Comment (danewalton) [reply sdk/samples/iot/paho_iot_pnp_sample.c:365]: yea I think I have to update this sample. My brain wasn't looking for those here but I can change them

File: sdk/samples/iot/paho_iot_pnp_sample.c (hunk 1 above)
Comment (danewalton) [reply sdk/samples/iot/paho_iot_pnp_sample.c:713]: Yea I actually was looking to change the API name to drop the `s` but it aligns with methods. Arg

File: sdk/samples/iot/paho_iot_pnp_sample.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (momuno) [reply sdk/samples/iot/paho_iot_pnp_sample.c]: I'm fine with whichever you choose.  Just curious :)

**Pull request merged**